src/main/scala/io/qbeast/spark/delta/writer/RollupDataWriter.scala

```diff
@@ -92,7 +92,7 @@ object RollupDataWriter
     processStats(stats, statsTrackers, fileStatsTracker)
     filesAndStats
       .map(_._1)
-      .map(IndexFiles.toAddFile())
+      .map(IndexFiles.toAddFile(dataChange = true))
```

Comment:
Do we roll up during optimize calls, if so, should data change really be true?

Reply:
Rollup happens before any data is written. The line outlined here is executed in regular writing.